core/domain/rights_manager.py

```diff
@@ -959,15 +842,17 @@ def _release_ownership_of_activity(committer_id, activity_id, activity_type):
     Raise:
         Exception. The committer does not have release rights.
     """
-    if not Actor(committer_id).can_release_ownership(
-            activity_type, activity_id):
+    committer_role = user_services.get_user_role_from_id(committer_id)
```

Comment:
Here and elsewhere, is there a way you could just pass in the user object so that we don't need to reconstruct it? It might make sense to remove some of the other functions and call release_ownership_of_activity() directly, if that helps. It depends on the code paths and whether it makes sense to pass a user object through or not.

Generally I wouldn't expect to see a UserActionsInfo constructed at this level. In fact it should probably never be constructed except in the base controller -- and then it should be passed around as needed. In places where you only need the ID you can then do user.id or user.get_id().